File: Professional Image Modifier 5/[C#]-Professional Image Modifier 5/C#/ImageFunctions/Forms/FrmModificationType.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

using WeifenLuo.WinFormsUI.Docking;

namespace ImageFunctions.Forms
{
    public partial class FrmModificationType : DockContent
    {

        public delegate void SusanSelectedHandler();
        public event SusanSelectedHandler SusanSelected;
        public delegate void MoravecSelectedHandler();
        public event MoravecSelectedHandler MoravecSelected;
        public delegate void HarrisSelectedHandler();
        public event HarrisSelectedHandler HarrisSelected;
        public delegate void FASTSelectedHandler();
        public event FASTSelectedHandler FASTSelected;

        private enum Modifications : int
        {
            None,
            CornerDetection,
        };

        private enum Methods
        {
            None,
            Susan,
            Moravec,
            Harris,
            FAST
        };


        public void EnableModifications()
        {
            lbModification.Enabled = true;
        }

        public void DisableModifications()
        {
            lbModification.Enabled = false;
        }

        public FrmModificationType()
        {
            InitializeComponent();
            lbModification.Items.Add("Corner Detection");
        }

        protected override string GetPersistString()
        {
            return this.Text;

        }

        private void lbModification_SelectedItemChanged(object sender, EventArgs e)
        {
            lbMethods.Enabled = true;
            switch (lbModification.SelectedIndex)
            {
                case (int)Modifications.None:
                    break;
                case (int)Modifications.CornerDetection:
                    PopulateMethods(Modifications.CornerDetection);
                    break;
                default:
                    break;
            }
        }

        //// Populate the Modification Method list box.
        private void PopulateMethods(Modifications modifications)
        {
            switch (modifications)
            {
                case Modifications.None:
                    lbMethods.Enabled = false;
                    break;
                case Modifications.CornerDetection:
                    lbMethods.Items.Add("Susan Corner Detection"); // index 1
                    lbMethods.Items.Add("Moravec Corner Detection"); // index 2
                    lbMethods.Items.Add("Harris Corner Detection");
                    lbMethods.Items.Add("FAST Corner Detection");
                    break;
                default:
                    lbMethods.Enabled = false;
                    break;
            }
        }

        private void lbMethods_SelectedItemChanged(object sender, EventArgs e)
        {
            switch (lbMethods.SelectedIndex)
            {
                case (int)Methods.None:
                    break;
                case (int)Methods.Susan:
                    SusanSelected();
                    break;
                case (int)Methods.Moravec:
                    MoravecSelected();
                    break;
                case (int)Methods.Harris:
                    HarrisSelected();
                    break;
                case (int)Methods.FAST:
                    FASTSelected();
                    break;
                default:
                    break;
            }
        }
    }
}

File: Professional Image Modifier 5/[VB]-Professional Image Modifier 5/VB/ImageFunctions/Forms/FrmModificationType.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Imports WeifenLuo.WinFormsUI.Docking

Namespace ImageFunctions.Forms
	Partial Public Class FrmModificationType
		Inherits DockContent

		Public Delegate Sub SusanSelectedHandler()
		Public Event SusanSelected As SusanSelectedHandler
		Public Delegate Sub MoravecSelectedHandler()
		Public Event MoravecSelected As MoravecSelectedHandler
		Public Delegate Sub HarrisSelectedHandler()
		Public Event HarrisSelected As HarrisSelectedHandler
		Public Delegate Sub FASTSelectedHandler()
		Public Event FASTSelected As FASTSelectedHandler

		Private Enum Modifications As Integer
			None
			CornerDetection
		End Enum

		Private Enum Methods
			None
			Susan
			Moravec
			Harris
			FAST
		End Enum


		Public Sub EnableModifications()
			lbModification.Enabled = True
		End Sub

		Public Sub DisableModifications()
			lbModification.Enabled = False
		End Sub

		Public Sub New()
			InitializeComponent()
			lbModification.Items.Add("Corner Detection")
		End Sub

		Protected Overrides Function GetPersistString() As String
			Return Me.Text

		End Function

		Private Sub lbModification_SelectedItemChanged(ByVal sender As Object, ByVal e As EventArgs) Handles lbModification.SelectedItemChanged
			lbMethods.Enabled = True
			Select Case lbModification.SelectedIndex
				Case CInt(Modifications.None)
				Case CInt(Modifications.CornerDetection)
					PopulateMethods(Modifications.CornerDetection)
				Case Else
			End Select
		End Sub

		'// Populate the Modification Method list box.
		Private Sub PopulateMethods(ByVal modifications As Modifications)
			Select Case modifications
				Case ImageFunctions.Forms.FrmModificationType.Modifications.None
					lbMethods.Enabled = False
				Case ImageFunctions.Forms.FrmModificationType.Modifications.CornerDetection
					lbMethods.Items.Add("Susan Corner Detection") ' index 1
					lbMethods.Items.Add("Moravec Corner Detection") ' index 2
					lbMethods.Items.Add("Harris Corner Detection")
					lbMethods.Items.Add("FAST Corner Detection")
				Case Else
					lbMethods.Enabled = False
			End Select
		End Sub

		Private Sub lbMethods_SelectedItemChanged(ByVal sender As Object, ByVal e As EventArgs) Handles lbMethods.SelectedItemChanged
			Select Case lbMethods.SelectedIndex
				Case CInt(Methods.None)
				Case CInt(Methods.Susan)
					RaiseEvent SusanSelected()
				Case CInt(Methods.Moravec)
					RaiseEvent MoravecSelected()
				Case CInt(Methods.Harris)
					RaiseEvent HarrisSelected()
				Case CInt(Methods.FAST)
					RaiseEvent FASTSelected()
				Case Else
			End Select
		End Sub
	End Class
End Namespace
